core/src/main/java/com/chopshop166/chopshoplib/commands/FunctionalWaitCommand.java

```diff
@@ -0,0 +1,59 @@
+package com.chopshop166.chopshoplib.commands;
+
+import java.util.function.DoubleSupplier;
+
+import edu.wpi.first.util.sendable.SendableRegistry;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/**
+ * A command that does nothing but takes a specified amount of time to finish.
+ * Useful for CommandGroups. Can also be subclassed to make a command with an
+ * internal timer.
+ *
+ */
+public class FunctionalWaitCommand extends CommandBase {
+
+    /** Used to see how much time is elapsed. */
+    protected Timer timer = new Timer();
+
+    /** Returns the duration to wait. */
+    private final DoubleSupplier durationSupplier;
+
+    /** The duration to wait. gets set when the command is initialized. */
+    private double duration;
+
+    /**
+     * Creates a new FunctionalWaitCommand. This command will do nothing, and end
+     * after the duration that is supplied.
+     *
+     * @param duration a DoubleSupplier returning the time to wait, in seconds.
+     */
+    public FunctionalWaitCommand(final DoubleSupplier duration) {
```

Comment:
Is `duration` the best name you can use for the argument? You use `duration` of a different type as a class member